repo_validation/scripts/validate_file_endings.ps1

```diff
@@ -0,0 +1,260 @@
+    to ensure they end with a proper newline character sequence (CRLF). Files that end
+    abruptly without a newline will be reported as validation failures.
+    
+    The script automatically excludes dependency directories (deps/, dependencies/) to
+    avoid modifying third-party code.
+    
+    When the -Fix switch is provided, the script will automatically correct files
+    by appending a CRLF newline at the end.
+
+.PARAMETER RepoRoot
+    The root directory of the repository to validate.
+
+.PARAMETER Fix
+    If specified, automatically fix files that do not end with a newline.
+
+.EXAMPLE
+    .\validate_crlf_line_endings.ps1 -RepoRoot "C:\repo"
+    
+    Validates all source files and reports errors.
+
+.EXAMPLE
+    .\validate_crlf_line_endings.ps1 -RepoRoot "C:\repo" -Fix
+    
+    Validates and automatically fixes all source files that don't end with a newline.
+
+.NOTES
+    Returns exit code 0 if all files are valid (or were fixed), 1 if validation fails.
+    Dependencies in deps/ and dependencies/ directories are automatically excluded.
+#>
+
+param(
+    [Parameter(Mandatory=$true)]
+    [string]$RepoRoot,
+    
+    [Parameter(Mandatory=$false)]
+    [string]$ExcludeFolders = "deps,cmake",
+    
+    [Parameter(Mandatory=$false)]
+    [switch]$Fix
+)
+
+# Set error action preference
+$ErrorActionPreference = "Stop"
+
+Write-Host "========================================" -ForegroundColor Cyan
+Write-Host "File Ending Newline Validation" -ForegroundColor Cyan
+Write-Host "========================================" -ForegroundColor Cyan
+Write-Host "Repository Root: $RepoRoot" -ForegroundColor White
+Write-Host "Fix Mode: $($Fix.IsPresent)" -ForegroundColor White
+Write-Host ""
+
+# Define file extensions to check
+$extensions = @("*.h", "*.hpp", "*.c", "*.cpp", "*.cs")
+
+# Parse excluded directories (default: deps, cmake)
+$excludeDirs = $ExcludeFolders -split "," | ForEach-Object { $_.Trim() } | Where-Object { $_ -ne "" }
+
+Write-Host "Excluded directories: $($excludeDirs -join ', ')" -ForegroundColor White
+Write-Host ""
+
+# Initialize counters
+$totalFiles = 0
+$invalidFiles = @()
+$fixedFiles = @()
+$skippedFiles = 0
+
+# Get all source files in the repository
+Write-Host "Searching for source files..." -ForegroundColor Yellow
+Write-Host "Excluding directories: $($excludeDirs -join ', ')" -ForegroundColor Yellow
+Write-Host ""
+
+foreach ($extension in $extensions) {
+    $files = Get-ChildItem -Path $RepoRoot -Filter $extension -Recurse -File -ErrorAction SilentlyContinue
+    
+    foreach ($file in $files) {
+        # Check if file is in an excluded directory
+        $relativePath = $file.FullName.Substring($RepoRoot.Length).TrimStart('\', '/')
+        $isExcluded = $false
+        
+        foreach ($excludeDir in $excludeDirs) {
+            if ($relativePath -like "$excludeDir\*" -or $relativePath -like "$excludeDir/*") {
```

Comment:
>$relativePath -like "$excludeDir\*" -or $relativePath -like "$excludeDir/*" [](http://example.com/codeflow?start=16&length=75)

I thought powershell could just automagically use forward slash or backslash interchangeably so I would think it wouldn't need to check both...